src/main/java/uk/gov/ons/ctp/response/action/service/impl/ActionPlanJobServiceImpl.java

```diff
@@ -71,84 +68,64 @@ public ActionPlanJob findActionPlanJob(final UUID actionPlanJobId) {
   public List<ActionPlanJob> createAndExecuteAllActionPlanJobs() {
     List<ActionPlanJob> executedJobs = new ArrayList<>();
     actionPlanRepo.findAll().forEach(actionPlan -> {
-      ActionPlanJob job = new ActionPlanJob();
-      job.setActionPlanFK(actionPlan.getActionPlanPK());
-      job.setCreatedBy(CREATED_BY_SYSTEM);
-      job = createAndExecuteActionPlanJob(job, false);
-      if (job != null) {
-        executedJobs.add(job);
+      Date lastExecutionTime = new Date(nowUTC().getTime() - appConfig.getPlanExecution().getDelayMilliSeconds());
+      if (actionPlan.getLastRunDateTime() == null || actionPlan.getLastRunDateTime().before(lastExecutionTime)) {
+        ActionPlanJob job = ActionPlanJob.builder()
```

Comment:
Creating the ActionPlanJob should be in the method createAndExecuteActionPlanJob or that name should be changed to just executeActionPlanJob.

Reply:
You could either pass in the ActionPlan or the actionplan pk as the parameter to createAndExecuteActionPlanJob as the only value used off this template is the job.actionPlanFk, which is actually straight off the action plan.

This would make it easier to understand as it's confusing seeing a created actionplanjob passed into a function called createAndExecuteActionPlanJob.